alternator/server.cc

```diff
@@ -208,7 +208,8 @@ class local_nodelist_handler : public gated_handler {
         // just the list of live nodes in this DC needs more elaborate code:
         auto& topology = _proxy.get_token_metadata_ptr()->get_topology();
         sstring local_dc = topology.get_datacenter();
-        std::unordered_set<gms::inet_address> local_dc_nodes = topology.get_datacenter_endpoints().at(local_dc);
```

Comment:
Note: the uncaptured get_token_metadata_ptr() above is surely a bug. @nyh 

The type/function for get_token_metadata_ptr() must be changed, it lulls people into a false sense of simplicity.

Reply:
> Note: the uncaptured get_token_metadata_ptr() above is surely a bug.

I'm not seeing it. What do you mean?

Reply:
it's embedded in `auto& topology = _proxy.get_token_metadata_ptr()->get_topology();`

Reply:
> it's embedded in auto& topology = _proxy.get_token_metadata_ptr()->get_topology();

Yes. What I meant is that the reference isn't used across preemption points, so I didn't see where the bug is. Avi clarified in another comment that it's not a bug now, but a bug waiting to happen.